aws/network-primitives/bootstrap_node_provisioner.tf

```diff
@@ -0,0 +1,153 @@
+
+  # prune network
+  provisioner "remote-exec" {
+    inline = [
+      "sudo chmod +x /tmp/prune_docker_system.sh",
+      "sudo /tmp/prune_docker_system.sh"
+    ]
+  }
+}
+
+resource "null_resource" "start-boostrap-nodes" {
+  count = length(local.bootstrap_nodes_ip_v4)
+
+  depends_on = [null_resource.setup-bootstrap-nodes]
+
+  # trigger on node deployment version change
+  triggers = {
+    deployment_version = var.bootstrap-node-config.deployment-version
+    reserved_only      = var.bootstrap-node-config.reserved-only
+  }
+
+  connection {
+    host        = local.bootstrap_nodes_ip_v4[count.index]
+    user        = var.ssh_user
+    type        = "ssh"
+    agent       = true
+    private_key = file("${var.private_key_path}")
+    timeout     = "300s"
+  }
+
+  # copy bootstrap node keys file
+  provisioner "file" {
+    source      = "./bootstrap_node_keys.txt"
+    destination = "/subspace/node_keys.txt"
+  }
+
+  # copy DSN bootstrap node keys file
+  provisioner "file" {
+    source      = "./dsn_bootstrap_node_keys.txt"
+    destination = "/subspace/dsn_bootstrap_node_keys.txt"
+  }
+
+  # copy compose file creation script
+  provisioner "file" {
+    source      = "${var.path_to_scripts}/create_bootstrap_node_compose_file.sh"
+    destination = "/subspace/create_compose_file.sh"
+  }
+
+  # start docker containers
+  provisioner "remote-exec" {
+    inline = [
+      # stop any running service
+      "sudo docker compose -f /subspace/docker-compose.yml down ",
+
+      # set hostname
+      "sudo hostnamectl set-hostname ${var.network_name}-bootstrap-node-${count.index}",
+
+      # create .env file
+      "echo NODE_ORG=${var.bootstrap-node-config.docker-org} > /subspace/.env",
+      "echo NODE_TAG=${var.bootstrap-node-config.docker-tag} >> /subspace/.env",
+      "echo NETWORK_NAME=${var.network_name} >> /subspace/.env",
+      "echo NODE_ID=${count.index} >> /subspace/.env",
+      "echo NODE_KEY=$(sed -nr 's/NODE_${count.index}_KEY=//p' /subspace/node_keys.txt) >> /subspace/.env",
+      "echo DATADOG_API_KEY=${var.datadog_api_key} >> /subspace/.env",
+      "echo PIECE_CACHE_SIZE=${var.piece_cache_size} >> /subspace/.env",
+      "echo DSN_NODE_KEY=$(sed -nr 's/NODE_${count.index}_KEY=//p' /subspace/dsn_bootstrap_node_keys.txt) >> /subspace/.env",
+      "echo DSN_LISTEN_PORT=${var.bootstrap-node-config.dsn-listen-port} >> /subspace/.env",
+      "echo NODE_DSN_PORT=${var.bootstrap-node-config.node-dsn-port} >> /subspace/.env",
+      "echo GENESIS_HASH=${var.bootstrap-node-config.genesis-hash} >> /subspace/.env",
+
+      # create docker compose file
+      "sudo chmod +x /subspace/create_compose_file.sh",
+      "sudo /subspace/create_compose_file.sh ${var.bootstrap-node-config.reserved-only} ${length(local.bootstrap_nodes_ip_v4)} ${count.index}",
+
+      # start subspace node
+      "sudo chown ubuntu:ubuntu /subspace/docker-compose.yml",
+      "sudo docker compose -f /subspace/docker-compose.yml up -d --build ",
```

Comment:
I'd prefer to not use systemd with terraform since it can break the run and would rather use systemd with configuration management. The --build flag just forces a rebuild of the images, not really applicable here since we don't use the `latest` tag, but it just forces a rebuild of the images before running them.

Reply:
If there is nothing to rebuild, it should not specify `--build`.

As to using systemd or not, since we use systemd right now, this line is incorrect since it will start container without systemd and will not apply iptables rules as the result.

Reply:
I thought we are deploying the same thing we had with minimal changes to AWS. What you're describing is quite far from it. Moreover, you're saying it is not the way to go, yet this PR still deploys the same systemd service and iptables rules, but in contrast to deployment to Hetzner where everything was done properly and in correct order, the way you wrote it is not equivalent, hence this comment and request to correct it to the way it was before.

The rules will certainly have an effect. They are probably redundant, but I have not visibility into what you have actually changed here comparing to what we had before unless I check out code locally and do file comparison manually.

> Thirdly, the current rules do nothing since they are targeting the wrong chain and should be targeting DOCKER-USER

They were certainly working properly and were not related to Docker as such at all. Our servers were blocked on Hetzner because we were making requests to private IP ranges via public interfaces, that is what those rules were blocking and they certainly did the job.